module_fv3_config.F90

```diff
@@ -28,5 +28,7 @@ module module_fv3_config
   character(esmf_maxstr),dimension(:),allocatable :: filename_base
   character(17)            :: calendar='                 '
   integer                  :: calendar_type = -99
+!
+  integer                  :: diagnostic = 0
```

Comment:
All variables in this module are read in from `model_configure` esmf config file, except `diagnostic` which is stored in fv3 component attributes. This is inconsistent.

Reply:
Hi @DusanJovic-NOAA 
I took a look through the fv3_cap and found an existing diagnostic routine that I'd like to mimic the `diagnose_cplFields` subroutine. So basically I would move `subroutine write_grid_netcdf` from `io/module_write_netcdf.F90` to `cpl/module_cap_cpl.F90` and rename it as `subroutine diagnose_fcstGrid`. I can eliminate the newly added `diagnostic` integer and utilize the existing diagnostic configuration variables. At the end of the `InitializeRealize` I would add the following.
```
if( dbug > 0 .or. cplprint_flag ) then
  call diagnose_fcstGrid(fcstGrid, "diagnostic_FV3_fcstGrid.nc", regridArea=.TRUE., rc=rc)
  if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) return
endif
```
I like this strategy because it fully follows existing diagnostic code built into FV3. What do you think?

Reply:
Yes, I think `cpl/module_cap_cpl.F90` is better place for `diagnose_fcstGrid` than `io/module_write_netcdf.F90` since it is using ESMF's write routines, while the code in module_write_netcdf is using native netcdf calls.

Reply:
This comment has been addressed in https://github.com/NOAA-EMC/fv3atm/pull/355/commits/6ac48b1d36b5559500dcb4718bae9140206fc2b8
I was unable to move the call into fv3_cap because the fcstGrid is created as a component of the module_fcst_grid_comp, which is a subset of the fv3_cap. I moved the subroutine out of io and into module_fcst_grid_comp and it is called using cplprint_flag (DumpFields)